### DISCUSSION TITLE
Use releases for internal builds

### DISCUSSION THREAD

**Discussion by jlbaptista** [actions/virtual-environments#8476] [General questions] [answered]
Hi.

We use your releases (tags) to create our internal build images. We use the releases so we can have the base image version match the ones used on Microsoft hosted agents.
I read on #8206 that, releases (tags) are not recommended to be used on internal builds.
To me, that does not make a lot sense, as my expectation is that the code on releases should be stable.

Please explain why releases should not be used on internal builds?

**Comment by vpolikarpov-akvelon** [comment, marked as answer]
Hey there. Let me break it down for you.

While releases (tags) might seem like a good fit for your internal builds because they give off that stable vibe, they're actually not the best choice for a few reasons.

First off, we create release branches and tags as markers in our ongoing process of updating images. They're not meant for regular end-user consumption. Our images have lots of software bits inside, and we don't lock them down to specific versions. Every time an image is built, it grabs the latest versions of the software to keep things up to date. Even if you use the same release branch or tag, you might end up with different software versions.

Also, our image generation scripts tend to become outdated pretty fast because the software landscape keeps changing. That's why the main branch is where things stay more stable.

Hope this clears things up a bit! If you have more questions or need further info, feel free to ask. We're here to help.

**Comment by jlbaptista** [comment]
Thanks @vpolikarpov-akvelon . 
It has indeed clarified things for me. We'll adjust our automation to always source from `main` branch. 
Thanks for the help.
